staging/src/kubevirt.io/client-go/kubecli/vm.go

```diff
@@ -179,7 +190,7 @@ func (v *vm) Restart(name string, restartOptions *v1.RestartOptions) error {
 		return fmt.Errorf(cannotMarshalJSONErrFmt, err)
 	}
 	uri := fmt.Sprintf(vmSubresourceURLFmt, v1.ApiStorageVersion, v.namespace, name, "restart")
-	return v.restClient.Put().RequestURI(uri).Body(body).Do(context.Background()).Error()
+	return v.restClient.Put().RequestURI(v.adaptUriForHostPath(uri)).Body(body).Do(context.Background()).Error()
```

Comment:
Why does the `base` of `restClient` don't safe as here?

Reply:
I don't understand your question @xpivarc sorry. Can you clarify ?

Reply:
I was looking into how the uri is constructed and I think the `base`(attribute of restClient) should cover this case and we would not need this special handling. I did not have time to check.
@vasiliy-ul wdyt?

Reply:
Hard to tell by looking just at the code... But isn't it a private attribute?

Reply:
Yes. It is probably populated from kubeconfig. That's why I am not sure we need this workaround. We should have a look at it as it would be easier than keeping this patch.

Reply:
Not sure if it can work with `base` from `RESTClient` since in the code we overwrite the URI for each request with `RequestURI(...)`. I guess this is needed to set the proper VM/VMI subresource path (which is not included in the `base` attribute as kubeconfig provides only the 'Host' of the base URI).

Reply:
UPD: hm... but maybe you are right. if we drop `RequestURI(...)` and use smth like `SubResource(...)` instead this will probably work.